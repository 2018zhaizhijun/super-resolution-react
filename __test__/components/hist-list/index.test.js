import React from 'react';
import renderer from 'react-test-renderer';
// import { act } from 'react-dom/test-utils';
// import { mount } from 'enzyme';
import HistList from '@components/hist-list';
// import MockAdapter from 'axios-mock-adapter';
// import axiosInstance from '@service/request';
//import toJson from "enzyme-to-json";

//let axiosMock = new MockAdapter(axiosInstance);

describe('actions about showing and deleting history items', () => {
    const list = [
        {
            'id': '1.jpg',
            'rawMaterial': 'mock/rawMaterial',
            'result': 'mock/result',
            'type': '0'
        },
        {
            'id': '2.mp4',
            'rawMaterial': 'mock/rawMaterial',
            'result': 'mock/result',
            'type': '1'
        }
    ]

    it('showing history list should be done correctly', () => {
        const histWrapper = renderer.create(
            <HistList initialList={list} />
        ).toJSON();
        
        expect(histWrapper).toMatchSnapshot();
    })

    // it('deleting history item should be done correctly', () => {
    //     const mockDeleteData = {
    //         'statusCode': 200,
    //         'errmsg': '成功',
    //         'data': {}
    //     }
          
    //     axiosMock.onDelete('/history').reply(200, mockDeleteData)
    //     let histWrapper = null;
    //     act(() => {
    //         histWrapper = mount(
    //             <HistList initialList={list} />
    //         );
    //         histWrapper.find('.anticon-close').first().simulate('click');
    //     })
    //     // click的callback函数包含异步请求，会在其执行完之前执行expect
    //     let updateList = histWrapper.find('Card');
    //     expect(updateList.length).toEqual(1);
    //     expect(updateList.first().props.key).toEqual('2.mp4');
    // })

})