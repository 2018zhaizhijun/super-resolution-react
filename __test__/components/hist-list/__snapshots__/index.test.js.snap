// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions about showing and deleting history items showing history list should be done correctly 1`] = `
<div>
  <div>
    <span
      aria-label="close"
      className="anticon anticon-close"
      onClick={[Function]}
      role="img"
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
    <div
      className="ant-card ant-card-bordered ant-card-contain-grid"
    >
      <div
        className="ant-card-head"
        style={Object {}}
      >
        <div
          className="ant-card-head-wrapper"
        >
          <div
            className="ant-card-head-title"
          >
            日期：
          </div>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-grid"
        >
          <img
            alt="raw image"
            src="http://101.35.24.184mock/rawMaterial"
          />
          <div
            className="conceal"
            onClick={[Function]}
          >
            <span>
              下载
            </span>
          </div>
        </div>
        <div
          className="ant-card-grid"
        >
          <img
            alt="processed image"
            src="http://101.35.24.184mock/result"
          />
          <div
            className=""
            onClick={[Function]}
          >
            <span>
              下载
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <span
      aria-label="close"
      className="anticon anticon-close"
      onClick={[Function]}
      role="img"
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
    <div
      className="ant-card ant-card-bordered ant-card-contain-grid"
    >
      <div
        className="ant-card-head"
        style={Object {}}
      >
        <div
          className="ant-card-head-wrapper"
        >
          <div
            className="ant-card-head-title"
          >
            日期：
          </div>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-grid"
        >
          <video
            controls="controls"
            src="http://101.35.24.184mock/rawMaterial"
          >
            该浏览器不支持 video 播放，请下载后观看
          </video>
          <div
            className="conceal"
            onClick={[Function]}
          >
            <span>
              下载
            </span>
          </div>
        </div>
        <div
          className="ant-card-grid"
        >
          <video
            controls="controls"
            src="http://101.35.24.184mock/result"
          >
            该浏览器不支持 video 播放，请下载后观看
          </video>
          <div
            className=""
            onClick={[Function]}
          >
            <span>
              下载
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
